### DISCUSSION TITLE
Everything you need to run v86 locally

### DISCUSSION THREAD

**Discussion by hello-smile6** [copy/v86#577] [Show and tell]
I'm hosting everything needed to run v86 on my `personal-server` Replit repl. Here's an example of using it which will work from a file on your computer without a webserver (Thanks, @copy , for answering #570 ).
```html
<!doctype html>
<title>Asynchronous loading of disk images</title>

<script src="https://personal-server.9pfs.repl.co/libv86.js"></script>
<script>
"use strict";

window.onload = function()
{
    // Async loading of the iso image
    // Note how the emulation starts without downloading the 50MB image

    // Support of the "Range: bytes=..." header is required on the server, CORS
    // is required if the server is on a different host

    var emulator = new V86Starter({
        wasm_path: "https://personal-server.9pfs.repl.co/v86.wasm",
        memory_size: 128 * 1024 * 1024,
        vga_memory_size: 8 * 1024 * 1024,
        screen_container: document.getElementById("screen_container"),
        bios: {
            url: "https://personal-server.9pfs.repl.co/seabios.bin",
        },
        vga_bios: {
            url: "https://personal-server.9pfs.repl.co/vgabios.bin",
        },
        cdrom: {
            url: "https://raw.githubusercontent.com/copy/images/db92d8fd4a53dbd73b8e0588bb4bd8fe37a1975a/linux.iso",
            async: false,
            // size can be determined automatically, but costs an extra request
            // and might not work reliably
        },
        autostart: true,
    });
}
</script>

<div id="screen_container">
    <div style="white-space: pre; font: 14px monospace; line-height: 14px"></div>
    <canvas style="display: none"></canvas>
</div>
```

**Comment by hello-smile6** [comment]
Apparently raw.githubusercontent.com sends cors headers, so why not abuse that and download from the source?

**Comment by hello-smile6** [comment]
Is it weird that Flappy bird runs better on v86 than an online Flappy bird game? Maybe there's some truth to my theory that v86 will replace Flash and become the new `#1` ad library. 1 ISO, for your kiosk displays, TV ads, and web ads. All you need.

**Reply by Darin755** [reply]
That is weird

**Comment by SugarRayLua** [comment]
@hello-smile6, could you code above to run v86 locally be modified to load a state rather (e.g. arch_state.bin.zst) rather than from a cd-rom iso? If so, how would one do so? I'm a novice linux/javascript user and have been exploring different ways to use v86 and would like to do by running it all locally on my device like you suggested but to start with the ArchLinux state first. Thanks!
